crates/kv/src/implementors/redis.rs

```diff
@@ -0,0 +1,62 @@
+use anyhow::{Context, Result};
+use redis::{Client, Commands};
+use slight_common::BasicState;
+
+/// This is the underlying struct behind the `AzBlob` variant of the `KvImplementor` enum.
+///
+/// It provides a property that pertains solely to the azblob implementation
+/// of this capability:
+///     - `container_client`
+///
+/// As per its' usage in `KvImplementor`, it must `derive` `Debug`, and `Clone`.
+#[derive(Debug, Clone)]
+pub struct RedisImplementor {
+    client: Client,
+}
+
+impl RedisImplementor {
+    pub fn new(slight_state: &BasicState, _name: &str) -> Self {
+        let connection_string = String::from_utf8(
+            slight_runtime_configs::get(
+                &slight_state.secret_store,
+                "REDIS_ADDRESS",
+                &slight_state.slightfile_path,
+            )
+            .with_context(|| {
+                format!(
+                    "failed to get 'REDIS_ADDRESS' secret using secret store type: {}",
+                    slight_state.secret_store
+                )
+            })
+            .unwrap(),
+        )
+        .unwrap();
```

Comment:
could you refactor this to use `get_from_state(.)` instead?

Reply:
sure